src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -512,4 +549,22 @@ CHIP_ERROR OTARequestor::SendApplyUpdateRequest(OperationalDeviceProxy & deviceP
     return cluster.InvokeCommand(args, this, OnApplyUpdateResponse, OnApplyUpdateFailure);
 }
 
+CHIP_ERROR OTARequestor::SendNotifyUpdateAppliedRequest(OperationalDeviceProxy & deviceProxy)
+{
+    VerifyOrReturnError(Basic::Attributes::SoftwareVersion::Set(kRootEndpointId, mUpdateVersion) == EMBER_ZCL_STATUS_SUCCESS,
```

Comment:
I don't think we should do that. In most cases the `NotifyUpdateApplied` will be sent after restarting the app/device so we won't have information about the new version in `mUpdateVersion`. Moreover, after restarting the app, it should initialize Basic cluster's Software Version automatically, so the correct value should already be in there. If you need it for the dummy implementation for Linux I suggest moving this line to Linux `OTAImageProcessorImpl::Apply`.

Reply:
On second thought, there are some more things about software version that we are not handling in OTA Requestor. I've documented them at https://github.com/project-chip/connectedhomeip/issues/13604 and can handle that in a separate PR. For the time being, I'll remove this call to update Basic cluster attribute and perhaps the right place to update this attribute is in a new API in OTARequestorInterface.